torchtitan/experiments/deepseek_v3/model.py

```diff
@@ -1189,3 +1229,11 @@ def _reorder_cache(past_key_values, beam_idx):
                 ),
             )
         return reordered_past
+
+    # Setup Symmetric Memory for MoE token shuffle.
+    # Supports inference currently.
```

Comment:
doesn't support training you mean?
there was a comment on the first PR in the stack that points out a bug in backward formula for all2all.  Otoh I thought training was supported.

Reply:
Yeah, I put a comment in the code:
```
For training, this is disabled for now. Reason is that autograd requires tensors not be modified a second time, this conflicts with our wish of sharing the symm mem across layers and/or PP microbatches.
```
Till we find a workaround, this is disabled for backward.

Reply:
We noticed this as well when debugging the backward pass / the OOM mentioned in #979. 

One way we were able to get around both issues is passing the symmetric memories to `on_device_all_to_all_v` and doing e.g. `token_gather_buffer.copy_(tensor)` inside the `forward` pass and changing `backward` accordingly. But tbh I'm not sure this resulted in correct gradients...

Reply:
@EugenHotaj Yeah, we may need to go the way you said.